presto-main/src/main/java/com/facebook/presto/server/TaskResource.java

```diff
@@ -326,4 +345,98 @@ private static Duration randomizeWaitTime(Duration waitTime)
         long halfWaitMillis = waitTime.toMillis() / 2;
         return new Duration(halfWaitMillis + ThreadLocalRandom.current().nextLong(halfWaitMillis), MILLISECONDS);
     }
+
+    // implement it as a static inner class as a simpler option to have a different URL path for the client download endpoint
+    @Path("/v1/download")
+    public static class TaskDownloadResource
+    {
+        private static final DataSize DEFAULT_MAX_SIZE = new DataSize(1.2, MEGABYTE);
+        private final TaskResource taskResource;
+
+        @Inject
+        public TaskDownloadResource(TaskResource taskResource)
+        {
+            this.taskResource = requireNonNull(taskResource, "taskResource is null");
+        }
+
+        @GET
+        @Path("{taskId}/results/{bufferId}/{token}")
```

Comment:
It's new in this class, because the logic is different: convert to json vs convert to pages. It uses the same approach with `RowIterable` as `StatementClient`.